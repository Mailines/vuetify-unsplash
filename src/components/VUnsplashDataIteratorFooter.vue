<template>
  <v-layout class="mt-2 mx-6 pa-2 d-flex justify-center align-center">
    <span class="grey--text mt-1">{{ label.itemsPerPage }}</span>
    <v-menu offset-y>
      <template v-slot:activator="{ on, attrs }">
        <v-btn
          :dark="isWhiteTextOnAccent"
          text
          :color="accentColor"
          class="ml-2"
          v-bind="attrs"
          v-on="on"
        >
          {{ imagesPerPage }}
          <v-icon>mdi-chevron-down</v-icon>
        </v-btn>
      </template>
      <v-list>
        <v-list-item
          v-for="(number, index) in itemsPerPageArray"
          :key="index"
          @click="updateItemsPerPage(number)"
        >
          <v-list-item-title>{{ number }}</v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
    <v-spacer></v-spacer>
    <div class="d-flex justify-center">
      <small class="grey--text" :style="dark ? { color: '#A6A6A6' } : {}"
        >{{ label.poweredBy }}
        <a
          href="https://unsplash.com/"
          style="text-decoration: none;"
          class="grey--text font-weight-bold"
          >Unplash</a
        ></small
      >
    </div>
    <v-spacer></v-spacer>
    <v-btn
      icon
      :color="accentColor"
      :disabled="page === 1"
      class="mr-1"
      @click="formerPage"
      id="former-page-btn-v-unsplash"
    >
      <v-icon>mdi-chevron-left</v-icon>
    </v-btn>
    <span
      class="mx-2 page-span-vunsplash
            grey--text"
    >
      {{ label.page }} {{ page }} {{ label.of }}
      {{ numberOfPages }}
    </span>

    <v-btn
      icon
      :color="accentColor"
      :disabled="page === numberOfPages"
      class="ml-1"
      @click="nextPage"
      id="next-page-btn-v-unsplash"
    >
      <v-icon>mdi-chevron-right</v-icon>
    </v-btn>
  </v-layout>
</template>
<script>
export default {
  props: {
    label: {
      type: Object,
      required: true
    },
    accentColor: {
      type: String,
      required: true
    },
    dark: {
      type: Boolean,
      required: true
    },
    isWhiteTextOnAccent: {
      type: Boolean,
      required: true
    },
    numberOfPages: {
      type: Number,
      required: true
    },
    imagesPerPage: {
      type: Number,
      required: true
    },
    page: {
      type: Number,
      required: true
    }
  },

  data() {
    return {
      itemsPerPageArray: [5, 10, 15, 20, 25, 30]
    };
  },
  methods: {
    updateItemsPerPage(number) {
      this.$emit("updateItemsPerPage", number);
    },
    formerPage() {
      this.$emit("formerPage");
    },
    nextPage() {
      this.$emit("nextPage");
    }
  }
};
</script>
