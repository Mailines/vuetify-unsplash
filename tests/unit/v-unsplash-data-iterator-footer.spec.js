import VUnsplashDataIteratorFooter from "../../src/components/VUnsplashDataIteratorFooter";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";

describe("VUnsplashDataIteratorFooter.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashDataIteratorFooter, {
      localVue,
      vuetify,
      propsData: {
        label: {
          itemsPerPage: "Items per page",
          page: "Page",
          of: "of",
          poweredBy: "powered by"
        },
        dark: true,
        numberOfPages: 4,
        imagesPerPage: 20,
        page: 2,
        accentColor: "#1976d2",
        isWhiteTextOnAccent: true
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("emits number of items per page event when click on list item of dropdown menu", async () => {
    wrapper.vm.updateItemsPerPage(5);
    expect(wrapper.emitted().updateItemsPerPage[0][0]).toBe(5);
  });
  it("emits former page event when clicking on former page button", async () => {
    wrapper.find("#former-page-btn-v-unsplash").trigger("click");
    expect(wrapper.emitted().formerPage).toBeTruthy();
  });
  it("emits nextPage event when clicking on next page button", async () => {
    wrapper.find("#next-page-btn-v-unsplash").trigger("click");
    expect(wrapper.emitted().nextPage).toBeTruthy();
  });
});
