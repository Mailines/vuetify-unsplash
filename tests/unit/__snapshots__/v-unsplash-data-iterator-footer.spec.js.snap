// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashDataIteratorFooter.vue renders and match snapshot 1`] = `
<div class="layout mt-2 mx-6 pa-2 d-flex justify-center align-center"><span class="grey--text mt-1">Items per page</span>
  <div class="v-menu">
    <!---->
  </div><button type="button" class="ml-2 v-btn v-btn--flat v-btn--text theme--dark v-size--default" style="color: rgb(25, 118, 210); caret-color: #1976d2;" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content">
        20
        <i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--dark"></i></span></button>
  <div class="spacer"></div>
  <div class="d-flex justify-center"><small class="grey--text" style="color: rgb(166, 166, 166);">powered by
      <a href="https://unsplash.com/" class="grey--text font-weight-bold" style="text-decoration: none;">Unplash</a></small></div>
  <div class="spacer"></div> <button type="button" class="mr-1 v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" style="color: rgb(25, 118, 210); caret-color: #1976d2;" id="former-page-btn-v-unsplash"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button> <span class="mx-2 page-span-vunsplash
          grey--text">
    Page 2 of
    4
  </span> <button type="button" class="ml-1 v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default" style="color: rgb(25, 118, 210); caret-color: #1976d2;" id="next-page-btn-v-unsplash"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button>
</div>
`;
